frontend/src/utils/hotkey_listener.ts

```diff
@@ -276,5 +284,98 @@ export class HotkeyListener {
     }
   }
 
+  private parseSpecialKeyBinding(binding: string) {
```

Comment:
Extended HotkeyListener with `onSpecialKeydown` and `onSpecialKeyup`. Also added some private helper methods.